DISTRO layer cleanup - Part A

hashraf91 opened the pull request:
* flex-os.conf: remove unused TOOLCHAINS_PATH variable
* flex-os.conf: remove unused psplash screen and its recipe
* flex-os.conf: remove unused DISTRO features by-default
* flex-os.conf: remove fix for meta-imx and meta-sca layers
* flex-os.conf: do not generate initramfs image by-default
* flex-os.conf: remove multilib-runtime image feature
* distro/support: remove multimedia, tools-audio and tools-benchmark
* production-image: remove the image recipe
* local.conf.sample: perform the cleanup
* distro/layer.conf: refresh dependencies on other layers

hashraf91 committed:
flex-os.conf: remove unused TOOLCHAINS_PATH variable

This is in continuation of bb6ba43dbdbfc37274d5e343a7e5a82918ae865d.
We now use internal yocto/oe-toolchain, so do not need to specify
the TOOLCHAINS_PATH.

JIRA-ID: SB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
flex-os.conf: remove unused psplash screen and its recipe

We are not using psplash anymore, hence removing it.

JIRA-ID: SB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
flex-os.conf: remove unused DISTRO features by-default

These DISTRO features are not being used by current BSPs, hence
removing them by-default from distro conf.

Also, removed acl from default DISTRO features.

Refs:
MentorEmbedded/meta-flex-ref-bsp@15a713d3:ana/meta-flex-bsp-ana-arm64/conf/distro/include/flex-distro.conf#L9
MentorEmbedded/meta-flex-ref-bsp@15a713d3:hycon/meta-flex-bsp-hycon-arm64/conf/distro/include/flex-distro.conf#L31

JIRA-ID: SB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
flex-os.conf: remove multilib-runtime image feature

This completely removes the multilib-runtime image feature:
* Removes the sdk_multilib_hook and image-sdk-multilib-variants
  classes
* Unlinks the sdk_extra_vars and sdk_codebench_metadata classes
  from sdk_multilib_hook
* Removes fixes from staging related to multilib feature which
  are not needed anymore
* Cleans-up multilib feature from flex-os.conf and local.conf

JIRA-ID: SB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
production-image: remove the image recipe

The production-image is no longer supported, hence removing its
recipe and related settings from local.conf.

Merged the flex-image.inc with development-image recipe as this
include is now only needed by development-image.

JIRA-ID: SB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
local.conf.sample: perform the cleanup

Remove IMAGE_DEVELOPMENT_FEATURES variable as we now only support
one image recipe and don't support production-image anymore. So, we
can directly add the image features through EXTRA_IMAGE_FEATURES.

Remove the explicit option to include ptest from local.conf.

Remove the examples to explain usage of EXTRA_IMAGE_FEATURES,
USER_FEATURES, package inclusion in target image and host/target
packages inclusion to SDK etc.

Moved the removal of nativesdk-packagegroup-sdk-host from local.conf
to flex-os.conf.

JIRA-ID: SB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

Noor-Ahsan commented on meta-flex-common/classes/sdk_extra_vars.bbclass:
```diff
@@ -2,15 +2,17 @@
-SDK_POSTPROCESS_MULTILIB_COMMAND += "add_sdk_extra_vars"
+# For convenience, in the commands
+REAL_MULTIMACH_TARGET_SYS ?= "${TUNE_PKGARCH}${TARGET_VENDOR}-${TARGET_OS}"
+SDK_ENV_SETUP_SCRIPT ?= "${SDK_OUTPUT}/${SDKPATH}/environment-setup-${REAL_MULTIMACH_TARGET_SYS}"
```
will it has any impact on SDK scripts. do we need to update anything to QA or documentation w.r,t this?

hashraf91 replied:
This won't have any impact on SDK scripts. This class was dependent on sdk_multilib_hook.bbclass. I have only moved the variables to this class and removed the dependency, so functionally nothing will change.

Noor-Ahsan commented on meta-flex-distro/conf/distro/flex-os.conf:
```diff
@@ -294,16 +276,12 @@ SDK_NAME_PREFIX = "${DISTRO}"
-SDK_TITLE ?= "${DISTRO_NAME} ${MLPREFIX_SECONDARY}${IMAGE_BASENAME} SDK for ${MACHINE}"
-SDK_TITLE:task-populate-sdk-ext:flex-os = "${DISTRO_NAME} ${MLPREFIX_SECONDARY}${IMAGE_BASENAME} Extensible SDK for ${MACHINE}"
+SDK_TITLE ?= "${DISTRO_NAME} ${IMAGE_BASENAME} SDK for ${MACHINE}"
+SDK_TITLE:task-populate-sdk-ext:flex-os = "${DISTRO_NAME} ${IMAGE_BASENAME} Extensible SDK for ${MACHINE}"
```
Kindly build SDK and see if there is any impact on CB integrations. does CB picks it up successfully?

hashraf91 replied:
This MLPREFIX_SECONDARY is empty variable in our case, and it was effective only for multilib feature to add the MLPREFIX.

hashraf91 committed:
flex-os.conf: do not generate initramfs image by-default

The current BSPs do not use initramfs image. This will be enabled
back if needed.

JIRA-ID: SB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
distro/layer.conf: refresh dependencies on other layers

meta-python is needed for python-unidiff which is being used by
python3_%.bbappend to add python3-patches.txt for release archives.

filesystem-layer and networking-layer are not direct dependencies
for our DISTRO layer now.

Hence, added dependency on meta-python and removed dependencies from
filesystem-layer and networking-layer.

JIRA-ID: BB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
flex-os.conf: remove fix for meta-imx and meta-sca layers

We do not support IMX BSPs now, so this can be removed.

JIRA-ID: SB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

ahsanhussain commented on meta-flex-support/recipes-core/images/production-image.bb:
```diff
@@ -1,13 +0,0 @@
-IMAGE_FEATURES_DISABLED_PRODUCTION ?= "debug-tweaks codebench-debug tools-profile"
-IMAGE_FEATURES:remove = "${IMAGE_FEATURES_DISABLED_PRODUCTION}"
-
-require recipes-core/images/flex-image.inc
```
@hashraf91 didn't we discuss merging the inc file in development image recipe, in this commit?

hashraf91 replied:
This will create conflicts for the second PR. However, I'll make this change as it makes sense to be included in the same commit.

ahsanhussain commented on meta-flex-distro/conf/templates/default/local.conf.sample:
```diff
@@ -14,26 +14,14 @@ MACHINE ??= "qemux86"
-IMAGE_FEATURES_DEVELOPMENT ?= "debug-tweaks"
-IMAGE_FEATURES_DEVELOPMENT:append:flex-bsp = " codebench-debug ssh-server-openssh"
-EXTRA_IMAGE_FEATURES = "${IMAGE_FEATURES_DEVELOPMENT}"
+EXTRA_IMAGE_FEATURES ?= "debug-tweaks codebench-debug ssh-server-openssh"
```
Hmm, I remember discussing to move this to development image recipe in our call. Weak assignment would ensure that users can add their own if they want.

hashraf91 replied:
Missed the part of moving it to development-image. Updated now.

hashraf91 committed:
distro/support: remove multimedia, tools-audio and tools-benchmark

These features are decommissioned. Removed these features along with
their recipes from distro/support.

JIRA-ID: SB-23686

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

ahsanhussain commented on meta-flex-support/recipes-core/images/development-image.bb:
```diff
@@ -5,11 +5,9 @@
-IMAGE_FEATURES_DEVELOPMENT:append:feature-tracing = " tools-profile"
-IMAGE_FEATURES_DISABLED_DEVELOPMENT ?= ""
-IMAGE_FEATURES:remove = "${IMAGE_FEATURES_DISABLED_DEVELOPMENT}"
+IMAGE_FEATURES = "${EXTRA_IMAGE_FEATURES}"
```
We don't have to do this manually, it's taken care of in the core layer. Just populate IMAGE_FEATURES here with a weak assignment so user can override those if needed.

ahsanhussain replied:
@Noor-Ahsan let's merge this PR and I'll fix this in #187
